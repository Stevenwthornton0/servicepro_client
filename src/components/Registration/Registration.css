.RegistrationForm > h2 {
    margin-bottom: 0;
    font-size: 36px;
}

.RegistrationForm > h3 {
    margin-top: 5px;
    font-weight: 300;
}

.RegistrationForm {
    width: 80%;
    font-weight: 400;
    margin-left: auto;
    margin-right: auto;
}

.RegistrationForm > div {
    margin-bottom: 20px;
}

.RegistrationForm > div > input {
    width: 100%;
    margin-top: 4px;
    border-radius: 4px;
    border: 1px solid #bdbdbd;
    font-size: 18px;
    padding: 8px;
}

.RegistrationForm > div > input:focus, .nameContainer > div > input:focus, .registerButton:focus {
    outline: none;
}

.first_name, .last_name {
    display: flex;
    flex-direction: column;
    width: 49%;
}

.registerButton {
    width: 100%;
    border: 1px solid lightblue;
    border-radius: 4px;
    font-size: 18px;
    font-weight: 400;
    padding-top: 10px;
    padding-bottom: 10px;
    background-color: lightblue;
}

.registerButton:hover {
    background-color: #424242;
    border: 1px solid #424242;
    color: lightblue;
    cursor: pointer;
}

.nameContainer {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.nameContainer > div > input {
    margin-top: 4px;
    font-size: 18px;
    padding: 8px;
    border: 1px solid #bdbdbd;
    border-radius: 4px;
}
.requires {
    line-height: 8px;
    font-size: 12px;
    color: #424242
}

@media screen and (min-width: 500px) {
    .RegistrationForm {
        width: 380px;
    }

    
}

