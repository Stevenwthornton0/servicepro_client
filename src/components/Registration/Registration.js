import React, { Component } from 'react';
import AuthApiService from '../../services/auth-api-service';
import './Registration.css';

class Registration extends Component {

    state = { error: null };

    static defaultProps = {
        onRegistrationSuccess: () => {},
    }

    handleSubmit = ev => {
        ev.preventDefault()
        const { first_name, last_name, user_name, password, email } = ev.target;

        this.setState({ error: null });
        AuthApiService.postUser({
            first_name: first_name.value,
            last_name: last_name.value,
            user_name: user_name.value,
            email: email.value,
            password: password.value,
            admin: 'false'
        })
            .then(user => {
                first_name.value = ''
                last_name.value = ''
                user_name.value = ''
                email.value=''
                password.value = ''
                this.props.onRegistrationSuccess()
            })
            .catch(res => {
                this.setState({error: res.error })
            })
    }

    render() {

        const { error } = this.state;
        return(
            <form
                className='RegistrationForm'
                onSubmit={this.handleSubmit}
            >
                <h2>Sign Up</h2>
                <h3>It's quick and easy.</h3>

                <div role='alert'>
                    {error && <p>{error}</p>}
                </div>
                <div className='nameContainer'>
                    <div className='first_name'>
                        <label htmlFor='RegistrationForm_first-name'>
                            First Name
                        </label>
                        <input
                            name='first_name'
                            type='text'
                            required
                            id='RegistrationForm_first-name'
                        />
                    </div>
                    <div className='last_name'>
                        <label htmlFor='RegistrationForm_last-name'>
                            Last Name
                        </label>
                        <input
                            name='last_name'
                            type='text'
                            required
                            id='RegistrationForm_last-name'
                        />
                    </div>
                </div>
                <div className='email'>
                    <label htmlFor='RegistrationForm_email'>
                        Email
                    </label>
                    <input
                        name='email'
                        type='email'
                        required
                        id='RegistrationForm_email'
                    />
                </div>
                <div className='user_name'>
                    <label htmlFor='RegistrationForm_user-name'>
                        User Name
                    </label>
                    <input
                        name='user_name'
                        type='text'
                        required
                        id='RegistrationForm_user-name'
                    />
                </div>
                <div className='pass_word'>
                    <label htmlFor='RegistrationForm_password'>
                        Password
                    </label>
                    <input
                        name='password'
                        type='password'
                        required
                        id='RegistrationForm_password'
                    />
                </div>
                <button type='submit' className='registerButton'>
                    Sign Up
                </button>
            </form>
        )
    }
}

export default Registration;