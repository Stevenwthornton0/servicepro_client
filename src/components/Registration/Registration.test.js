import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import Registration from './Registration';

Enzyme.configure({ adapter: new Adapter() })

describe('Registration component', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<Registration />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})