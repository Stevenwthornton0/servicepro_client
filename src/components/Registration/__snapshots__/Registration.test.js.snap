// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration component renders without crashing 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <h2>
    Sign Up
  </h2>
  <h3>
    It's quick and easy.
  </h3>
  <div
    className="red"
    role="alert"
  />
  <div
    className="nameContainer"
  >
    <div
      className="first_name"
    >
      <label
        htmlFor="RegistrationForm_first-name"
      >
        First Name
      </label>
      <input
        id="RegistrationForm_first-name"
        name="first_name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="last_name"
    >
      <label
        htmlFor="RegistrationForm_last-name"
      >
        Last Name
      </label>
      <input
        id="RegistrationForm_last-name"
        name="last_name"
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="email"
  >
    <label
      htmlFor="RegistrationForm_email"
    >
      Email
    </label>
    <input
      id="RegistrationForm_email"
      name="email"
      required={true}
      type="email"
    />
  </div>
  <div
    className="user_name"
  >
    <label
      htmlFor="RegistrationForm_user-name"
    >
      User Name
    </label>
    <input
      id="RegistrationForm_user-name"
      name="user_name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="pass_word"
  >
    <label
      htmlFor="RegistrationForm_password"
    >
      Password
    </label>
    <p
      className="requires computer"
    >
      Uppercase, lowercase, number & special character required
    </p>
    <input
      id="RegistrationForm_password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="requires mobile"
  >
    <p
      className="requires"
    >
      Requirements:
    </p>
    <p>
      • an uppercase letter
    </p>
    <p>
      • a lowercase letter
    </p>
    <p>
      • a number
    </p>
    <p>
      • a special character
    </p>
  </div>
  <button
    className="registerButton"
    type="submit"
  >
    Sign Up
  </button>
  <div
    className="loader"
  />
</form>
`;
